### PR TITLE
Bug 1824869 - Add PrivateBrowsing[Interactor|Controller] for handling private browsing mode

### PR DESCRIPTION
**Pull request by gabrielluong**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1824869

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/home/privatebrowsing/controller/PrivateBrowsingController.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.home.Mode
+import org.mozilla.fenix.home.privatebrowsing.interactor.PrivateBrowsingInteractor
+import org.mozilla.fenix.settings.SupportUtils
+
+/**
+ * An interface that handles the view manipulation of the private browsing mode.
+ */
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/home/privatebrowsing/controller/PrivateBrowsingController.kt:22]: is the long term goal for this interactor/controller to handle **all** private mode interactions in Fenix, or just the ones on the homepage? 

(just not sure if the file directory this resides in is the answer)

File: fenix/app/src/main/java/org/mozilla/fenix/home/privatebrowsing/controller/PrivateBrowsingController.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+/**
+ * The default implementation of [PrivateBrowsingController].
+ */
+class DefaultPrivateBrowsingController(
+    private val activity: HomeActivity,
+    private val appStore: AppStore,
+    private val navController: NavController,
+) : PrivateBrowsingController {
+
+    override fun handlePrivateBrowsingLearnMoreClicked() {
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/home/privatebrowsing/controller/PrivateBrowsingController.kt:44]:
```suggestion
    override fun handleLearnMoreClicked() {
```
after this refactor, does this contract function still need to reference that it's with regards to private browsing?

**Review by MozillaNoah — APPROVED** [review]
LGTM. One question and one function naming nit.

File: fenix/app/src/main/java/org/mozilla/fenix/home/privatebrowsing/controller/PrivateBrowsingController.kt (hunk 1 above)
Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/home/privatebrowsing/controller/PrivateBrowsingController.kt:22]: Just the Homepage as specified by the package name.

**Pull request merged**